frontera/contrib/backends/cassandra/models.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+import uuid
+from cassandra.cqlengine.models import Model
+from cassandra.cqlengine.usertype import UserType
+from cassandra.cqlengine.columns import Map, Text, Float, Integer, DateTime, UserDefinedType, Counter, Boolean, \
+    SmallInt, BigInt
+
+
+class Meta(UserType):
+    domain = Map(Text(), Text(), required=False)
+    fingerprint = Text()
+    origin_is_frontier = Boolean()
+    scrapy_callback = Text()
+    scrapy_errback = Text()
+    scrapy_meta = Map(Text(), Text(), required=False)
+    score = Float(required=False)
+    jid = Integer(required=False)
+
+
+class MetadataModel(Model):
+    __table_name__ = 'metadata'
+
+    crawl = Text(primary_key=True)
+    fingerprint = Text(primary_key=True)
+    url = Text(index=True)
+    depth = Integer()
+    created_at = DateTime()
+    fetched_at = DateTime(required=False)
+    status_code = Integer(required=False)
+    score = Float(required=False)
+    error = Text(required=False)
+    meta = UserDefinedType(Meta)
+    headers = Map(Text(), Text(), required=False)
+    cookies = Map(Text(), Text(), required=False)
+    method = Text(required=False)
+
+    @classmethod
+    def query(cls, session):
+        return session.query(cls)
+
+    def __repr__(self):
+        return '<Metadata:%s (%s)>' % (self.url, self.fingerprint)
+
+
+class StateModel(Model):
+    __table_name__ = 'states'
+
+    crawl = Text(primary_key=True)
+    fingerprint = Text(primary_key=True)
+    state = SmallInt(index=True)
+
+    @classmethod
+    def query(cls, session):
+        return session.query(cls)
+
+    def __repr__(self):
+        return '<State:%s=%d>' % (self.fingerprint, self.state)
+
+
+class QueueModel(Model):
+    __table_name__ = 'queue'
+
+    crawl = Text(primary_key=True)
```

Comment:
Hi Michael. I was thinking about data model you designed for Queue, and it's hard for me to see the logic there. What goals did you set? What access pattern (e.g. queries) you see? Is this designed for broad crawling?